packages/nextjs/components/scaffold-eth/TokenAmount.tsx

```diff
@@ -0,0 +1,22 @@
+type TokenAmountProps = {
+  amount?: number;
+  precision?: number;
+  currencyPosition?: "left" | "right";
+  currency?: string;
+};
+
+/**
+ * Display a token amount with a custom precision.
+ */
+export const TokenAmount = ({ amount = 0, precision = 4, currencyPosition = "left", currency }: TokenAmountProps) => {
+  const displayAmount = parseFloat(amount.toString()).toFixed(precision);
+  const currencySymbol = <span className="text-[0.8em] font-bold">{currency}</span>;
+
+  return (
+    <div className="flex items-center justify-center gap-1">
+      {currencyPosition === "left" && currencySymbol}
+      {displayAmount}
+      {currencyPosition === "right" && currencySymbol}
+    </div>
+  );
+};
```

Comment:
Umm not sure if this would be a great add or not :( 

I think the final version of this component will be something like clicking the component it will toggle and display `displayAmount * price` (`price` will be coming from prop) 

I think the toggling behavior can be easily achieved by https://daisyui.com/components/swap/

Also accepting `amount` as number not sure if its best way because usually while reading token amount you would get bigInt maybe people love to use `parseEther` / `parseUnits` to show it properly ?

Maybe I think it's better we let developer control all the styling too since its minimal, we can create a recipe in docs.scaffoldeth.io to help beginners if we see a lot of people struggling on this 🙌